dm/syncer/syncer.go

```diff
@@ -1621,11 +1632,13 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 			if s.streamerController.CanRetry(err) {
 				// lastLocation is the last finished GTID
-				err = s.streamerController.ResetReplicationSyncer(tctx, lastLocation)
+				err = s.streamerController.ResetReplicationSyncer(tctx, s.checkpoint.GlobalPoint())
```

Comment:
If this retry is occurred at `sharding-resyc` phase, enable safe-mode for one transaction is not enough because the start point may be reset to the very beginning point. But since safe-mode is enable in the full sharding-resync phase this is ok. Maybe we can add a comment about this here.

Reply:
What does `safe-mode is enable in the full sharding-resync phase` means? To me this is a problem.

Reply:
we will open safemode when there's a unsynced sharding table

https://github.com/pingcap/ticdc/blob/d381504a9fc3e5e6615e17627c424ec5bd17b0b4/dm/syncer/syncer.go#L2680

https://github.com/pingcap/ticdc/blob/d381504a9fc3e5e6615e17627c424ec5bd17b0b4/dm/syncer/syncer.go#L2708

please help check it

Reply:
Looks good to me